### PR TITLE
Migration guide for `ThemeData.dialogBackgroundColor` deprecation

### PR DESCRIPTION
**Pull request by TahaTesser**
Related to https://github.com/flutter/flutter/pull/155072

## Presubmit checklist

- [x] This PR is marked as draft with an explanation if not meant to land until a future stable release.
- [x] This PR doesn’t contain automatically generated corrections (Grammarly or similar).
- [x] This PR follows the [Google Developer Documentation Style Guidelines](https://developers.google.com/style) — for example, it doesn’t use _i.e._ or _e.g._, and it avoids _I_ and _we_ (first person).
- [x] This PR uses [semantic line breaks](https://github.com/dart-lang/site-shared/blob/main/doc/writing-for-dart-and-flutter-websites.md#semantic-line-breaks) of 80 characters or fewer.

### CODE REVIEW

**Comment by TahaTesser** [comment]
Changed the status of this PR, now that https://github.com/flutter/flutter/pull/155072 is landing.

File: src/content/release/breaking-changes/deprecate-themedata-dialogbackgroundcolor.md [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+  `DialogThemeData.backgroundColor`
+description: >-
+  The `ThemeData.dialogBackgroundColor` parameter has been replaced by
+  `DialogThemeData.backgroundColor`.
+---
+
+## Summary
+
+The [`ThemeData.dialogBackgroundColor`][] parameter was deprecated in favor of
+[`DialogThemeData.backgroundColor`][] parameter.
```
Comment (sfshaza2) [inline src/content/release/breaking-changes/deprecate-themedata-dialogbackgroundcolor.md:12]:
```suggestion
the [`DialogThemeData.backgroundColor`][] parameter.
```

File: src/content/release/breaking-changes/deprecate-themedata-dialogbackgroundcolor.md [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+## Context
+
+The defaults for the [`Dialog`][] and [`AlertDialog`][] widgets can be
+overridden with component-specific theme like [`DialogThemeData`][].
```
Comment (sfshaza2) [inline src/content/release/breaking-changes/deprecate-themedata-dialogbackgroundcolor.md:17]:
```suggestion
overridden with a component-specific theme like [`DialogThemeData`][].
```

File: src/content/release/breaking-changes/deprecate-themedata-dialogbackgroundcolor.md [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+Previously, the `ThemeData.dialogBackgroundColor` parameter was used to
+override the default dialog background color, which was made redundant by
+[`DialogThemeData`][].
+
+## Description of change
+
+The [`ThemeData.dialogBackgroundColor`][] is deprecated in favor of
+component-specific theme. Use [`DialogThemeData`][] to override the default
```
Comment (sfshaza2) [inline src/content/release/breaking-changes/deprecate-themedata-dialogbackgroundcolor.md:25]:
```suggestion
a component-specific theme. Use [`DialogThemeData`][] to override the default
```

**Review by sfshaza2 — APPROVED** [review]
lgtm, with a few grammatical fixes.

**Review by Piinks — APPROVED** [review]
LGTM

**Pull request merged**